### PR TITLE
Paradigm Restructuring Part 3 - Logger & HTTPClient

### PR DESCRIPTION
**Pull request by nupplaphil**
FollowUp #10919 
Addons: https://github.com/friendica/friendica-addons/pull/1198

- Restructuring Logger
- Restructuring HTTPClient

### CODE REVIEW

**Comment by nupplaphil** [comment]
I think this one can get merged as well
- Logger logs (Worker, normal app) like before
- I already fetched posts 
- I already see curl redirects inside the log

File: src/Core/Logger/Factory/Logger.php [hunk 1]
```diff
@@ -19,45 +19,45 @@
...
-	 * @var array
+	 * @var string[]
 	 */
 	private static $ignoreClassList = [
-		Logger::class,
+		Core\Logger::class,
 		Profiler::class,
-		'Friendica\\Util\\Logger',
+		'Friendica\\Core\\Logger\\Type',
+		'Friendica\\Core\\Logger\\Type\\Monolog',
```
Comment (MrPetovan) [inline src/Core/Logger/Factory/Logger.php]: Please use the `::class` magic constant for these:
```suggestion
		\Friendica\Core\Logger\Type::class,
		\Friendica\Core\Logger\Type\Monolog::class,
```

File: src/Core/Logger/Factory/Logger.php [hunk 2]
```diff
@@ -106,7 +106,8 @@ public function create(Database $database, IManageConfigValues $config, Profiler
 						static::addStreamHandler($logger, $stream, $loglevel);
 					} catch (\Throwable $e) {
 						// No Logger ..
-						$logger = new VoidLogger();
+						/// @todo isn't it possible to give the admin any hint about this wrong configuration?
```
Comment (MrPetovan) [inline src/Core/Logger/Factory/Logger.php]: I think we can default to `Syslog` instead?

File: src/Network/HTTPClient/Capability/ICanRequestPerHttp.php [hunk 3]
```diff
@@ -19,14 +19,14 @@
...
-namespace Friendica\Network;
+namespace Friendica\Network\HTTPClient\Capability;
 
 use GuzzleHttp\Exception\TransferException;
 
 /**
  * Interface for calling HTTP requests and returning their responses
  */
-interface IHTTPClient
+interface ICanRequestPerHttp
```
Comment (MrPetovan) [inline src/Network/HTTPClient/Capability/ICanRequestPerHttp.php]: "per"? It sounds odd to my American English-trained ear. How about one of these alternatives:
```suggestion
interface ICanPerformHttpRequests
```
```suggestion
interface ICanIssueHttpRequests
```
```suggestion
interface ICanSendHttpRequests
```
I
I'm partial to the latter, we already use the verb `send` in the phpDoc.

File: src/Network/HTTPClient/Client/HttpClientCan.php [hunk 4]
```diff
@@ -37,7 +42,7 @@
 /**
  * Performs HTTP requests to a given URL
  */
-class HTTPClient implements IHTTPClient
+class HttpClientCan implements ICanRequestPerHttp
```
Comment (MrPetovan) [inline src/Network/HTTPClient/Client/HttpClientCan.php]: `HttpClientCan`? Like a tin can 🥫?

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Core/Logger/Factory/Logger.php (hunk 1 above)
Comment (nupplaphil) [reply src/Core/Logger/Factory/Logger.php]: This works even if there is no class?

File: src/Network/HTTPClient/Capability/ICanRequestPerHttp.php (hunk 3 above)
Comment (nupplaphil) [reply src/Network/HTTPClient/Capability/ICanRequestPerHttp.php]: A little bit too denglish :D

File: src/Core/Logger/Factory/Logger.php (hunk 1 above)
Comment (MrPetovan) [reply src/Core/Logger/Factory/Logger.php]: Or, they are meant as prefixes? In this case the second line is superfluous as it is included in the more general prefix defined before.
```suggestion
		'Friendica\Core\Logger\Type',
```

File: src/Core/Logger/Factory/Logger.php (hunk 2 above)
Comment (nupplaphil) [reply src/Core/Logger/Factory/Logger.php]: I added it with another try/catch block :D .. The only real problem would be, if we set an invalid minimum level, so we would get argument exceptions at any case.

=> Shall I create a fallback min-level in that case?

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**